compiler/optimizer/OMRSimplifierHandlers.cpp

```diff
@@ -9205,31 +9249,58 @@ TR::Node *bdivSimplifier(TR::Node * node, TR::Block * block, TR::Simplifier * s)
    {
    s->simplifyChildren(node, block);
 
+   // Handshake with divchkSimplifier:  If simplifying the child of a DIVCHK
+   // results in a node that still needs to have a DIVCHK applied, place that
+   // node in _nodeToDivchk.  If the simplification leaves no node that needs
+   // to have a DIVCHK applied, set _nodeToDivchk to NULL.
+   s->_nodeToDivchk = node;
+
    TR::Node * firstChild = node->getFirstChild(), * secondChild = node->getSecondChild();
 
-   if (firstChild->getOpCode().isLoadConst() && secondChild->getOpCode().isLoadConst())
+   if (secondChild->getOpCode().isLoadConst()
+       && (secondChild->getByte() != 0)
+       && permitSimplificationOfConstantDivisor(s, node))
       {
-      foldByteConstant(node, (int8_t)(firstChild->getByte() / secondChild->getByte()), s, false /* !anchorChildren*/);
-      return node;
```

Comment:
With this return removed, a division of `c/1` for constant `c` will be first folded and then also replaced by `firstChild` in `BINARY_IDENTITY_OP()`. That struck me as odd, but AFAICT it should be harmless

Reply:
Thanks for catching that!  I'll fix it here and in `sdivSimplifier` as well.

Reply:
Fixed in commit f3b666df3ba1a0d59b2770123fd6ba14447b4bce